compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6356,9 +6360,24 @@ public void visit(BLangInvocation.BLangActionInvocation actionInvocation) {
         if (!actionInvocation.async && actionInvocation.invokedInsideTransaction) {
             transactionDesugar.startTransactionCoordinatorOnce(env, actionInvocation.pos);
         }
+
+        // Add `@strand {thread: "any"}` annotation to an isolated start-action or
+        // isolated named worker declaration.
+        if (actionInvocation.async && Symbols.isFlagOn(actionInvocation.symbol.type.flags, Flags.ISOLATED)) {
+            addStrandAnnotationWithThreadAny(actionInvocation);
+        }
+
         rewriteInvocation(actionInvocation, actionInvocation.async);
     }
 
+    private void addStrandAnnotationWithThreadAny(BLangInvocation.BLangActionInvocation actionInvocation) {
+        BLangPackage pkgNode = env.enclPkg;
+        List<BLangTypeDefinition> prevTypeDefinitions = new ArrayList<>(pkgNode.typeDefinitions);
+        annotationDesugar.addStrandAnnotationWithThreadAny(actionInvocation, env);
+        // Add record type node for internally added strand annotation.
+        addRecordTypeNodeForTypeDef(pkgNode, prevTypeDefinitions);
```

Comment:
Will we be adding a new type node/definition for each new strand annotation we add? Unnecessary, right? Since it's kinda a constant.

Reply:
What I meant so say was maybe we can have a predefined type/type definition that is reused?

Reply:
Fixed